mlflow/tracking/fluent.py

```diff
@@ -1308,20 +1309,27 @@ def print_auto_logged_info(r):
     }
 
     def setup_autologging(module):
-        autolog_fn = LIBRARY_TO_AUTOLOG_FN[module.__name__]
         try:
-            needed_params = list(inspect.signature(autolog_fn).parameters.keys())
-            filtered = {k: v for k, v in locals_copy if k in needed_params}
-        except ValueError:
-            filtered = {}
+            autolog_fn = LIBRARY_TO_AUTOLOG_FN[module.__name__]
+            try:
+                needed_params = list(inspect.signature(autolog_fn).parameters.keys())
+                filtered = {k: v for k, v in locals_copy if k in needed_params}
+            except ValueError:
```

Comment:
Restructuring this for safety. Previously, if the dictionary lookup failed or the `try` block threw a non-ValueError exception, `mlflow.autolog()` would break.

Reply:
Should we use `KeyError` here instead of `ValueError`?

```
>>> {}["foo"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'foo'
```

Reply:
wait

Reply:
which line causes `ValueError`?

Reply:
`inspect.signature` raises the `ValueError` (https://docs.python.org/3/library/inspect.html#inspect.signature). Though, according to the docs, it may also throw `TypeError`. To avoid headaches with exception typing later on, I've gone ahead and just changed this to a broad except.